### ISSUE TITLE
[1.3-beta.12] Can't import mocha

### ISSUE THREAD

**Issue by aldeed** [meteor/meteor#6430]
When you `npm i --save-dev mocha` and then `import Mocha from 'mocha'` in a Meteor 1.3-beta.12 app test file, you get the following error when doing `meteor test`:

```
Error: Can't find npm module 'lru-cache'. Did you forget to call 'Npm.depends' in package.js within the 'modules-runtime' package?
```

Which seems to stem from this line: https://github.com/isaacs/minimatch/blob/72845fa0454d03adf2bb551af35c36835878ff34/minimatch.js#L35

The same thing happens if you `Npm.depends({'mocha'})` in a package. So currently it's impossible to make a Meteor test driver package for Mocha without copying the Mocha source files into it. (cc @tmeasday)

This is a pretty old version of minimatch that mocha pkg uses, so I don't know if that's the issue.

**Comment by zol** [comment]
@benjamn Have you seen this before? I would be tempted to say let's bump this till post 1.3 if it wasn't so closely tied to our testing stuff.

**Comment by tmeasday** [comment]
Repro-ed

We don't *need* it to be resolved for 1.3 but @aldeed is right, it would be way better to require mocha in this way.

It's super weird how it happens. I think it might be worth figuring out what the issue is anyway. @benjamn do we have a document somewhere that explains how to go about debugging these kind of issues? That might be useful, then @zol and I (and others) could try and figure out the root cause in these cases...

**Comment by benjamn** [comment]
The culprit here is https://github.com/meteor/meteor/commit/99cb96fcf76cb08cc389d354ec8ed42ef74e8ae9, which means we don't pay attention to the [`require("lru-cache")`](https://github.com/isaacs/minimatch/blob/72845fa0454d03adf2bb551af35c36835878ff34/minimatch.js#L35) call in `minimatch.js`, because the file is wrapped in a [closure](https://github.com/isaacs/minimatch/blob/72845fa0454d03adf2bb551af35c36835878ff34/minimatch.js#L1) that shadows the global `require` variable, even though the value of that parameter happens to be [the same](https://github.com/isaacs/minimatch/blob/72845fa0454d03adf2bb551af35c36835878ff34/minimatch.js#L1051) as the global `require` function.

This will be fixed by #6398, but it could also be fixed by using a newer version of `minimatch`, as @aldeed suggested, since newer versions don't use a closure anymore.

**Comment by tmeasday** [comment]
It'll fix it on the server but I see the same issue on the client, which I don't imagine will be fixed?

**Comment by benjamn** [comment]
That's right. You may need to `npm install lru-cache` yourself and `import "lru-cache"` somewhere to ensure it's included in the client bundle.

We could also consider reverting https://github.com/meteor/meteor/commit/99cb96fcf76cb08cc389d354ec8ed42ef74e8ae9, but I feel like it's more helpful than harmful?

**Comment by tmeasday** [comment]
Well including `lru-cache` and `sigmund` in that way worked (on the client at least; on the server it had issues that I suspect will be solved by #6398).

@aldeed what this means is that (for now at least, maybe mocha will bump versions of that package / maybe we can PR them to?) we could get this working, (I guess you'd need to `Npm.depend` on those two packages?)

@benjamn I agree that it seems like this is a weird situation and that short-cut is pretty helpful.

**Comment by aldeed** [comment]
@tmeasday The outdated minimatch actually comes from an outdated glob dep, for which there is already a PR you can +1: https://github.com/mochajs/mocha/pull/2123

**Comment by aldeed** [comment]
This is resolved for me running the latest release-1.3 commit.
